### PR TITLE
update date command in Makefile to work crossplatform

### LINKED ISSUE #226: Make install doesn't work due to rfc date standard issue in make file
**What happened**:
worked on : ubuntu and windows but not on mac and alpine based images
**make install** fails due to date issue on alpine based docker images and mac due to not supported rfc-3339 for date 
**What you expected to happen**:
It should install ocm on inside the image .
**How to reproduce it (as minimally and precisely as possible)**:
docker file where I have checked 
check in Makefile line no 17 
** NOW         := $(shell date --rfc-3339=seconds | sed 's/ /T/')**
**Anything else we need to know**:
logs :
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/cespare/xxhash/v2 v2.1.2
go: downloading github.com/prometheus/procfs v0.7.3
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.4
date: unrecognized option: rfc-3339=seconds
BusyBox v1.35.0 (2022-11-19 10:13:10 UTC) multi-call binary.

Usage: date [OPTIONS] [+FMT] [[-s] TIME]

Display time (using +FMT), or set time

        -u              Work in UTC (don't convert to local time)
        [-s] TIME       Set time to TIME
        -d TIME         Display TIME, not 'now'
        -D FMT          FMT (strptime format) for -s/-d TIME conversion
        -r FILE         Display last modification time of FILE
        -R              Output RFC-2822 date
        -I[SPEC]        Output ISO-8601 date
                        SPEC=date (default), hours, minutes, seconds or ns

Recognized TIME formats:
        @seconds_since_1970
        hh:mm[:ss]
        [YYYY.]MM.DD-hh:mm[:ss]
        YYYY-MM-DD hh:mm[:ss]
        [[[[[YY]YY]MM]DD]hh]mm[.ss]
        'date TIME' form accepts MMDDhhmm[[YY]YY][.ss] instead
date: unrecognized option: rfc-3339=seconds
BusyBox v1.35.0 (2022-11-19 10:13:10 UTC) multi-call binary.

Usage: date [OPTIONS] [+FMT] [[-s] TIME]

Display time (using +FMT), or set time

        -u              Work in UTC (don't convert to local time)
        [-s] TIME       Set time to TIME
        -d TIME         Display TIME, not 'now'
        -D FMT          FMT (strptime format) for -s/-d TIME conversion
        -r FILE         Display last modification time of FILE
        -R              Output RFC-2822 date
        -I[SPEC]        Output ISO-8601 date
                        SPEC=date (default), hours, minutes, seconds or ns

Recognized TIME formats:
        @seconds_since_1970
        hh:mm[:ss]
        [YYYY.]MM.DD-hh:mm[:ss]
        YYYY-MM-DD hh:mm[:ss]
        [[[[[YY]YY]MM]DD]hh]mm[.ss]
        'date TIME' form accepts MMDDhhmm[[YY]YY][.ss] instead
^Cmake: *** [Makefile:68: install] Interrupt

/opt/ocm # make install > logs.txt
date: unrecognized option: rfc-3339=seconds
BusyBox v1.35.0 (2022-11-19 10:13:10 UTC) multi-call binary.

Usage: date [OPTIONS] [+FMT] [[-s] TIME]

Display time (using +FMT), or set time

        -u              Work in UTC (don't convert to local time)      
        [-s] TIME       Set time to TIME
        -d TIME         Display TIME, not 'now'
        -D FMT          FMT (strptime format) for -s/-d TIME conversion
        -r FILE         Display last modification time of FILE
        -R              Output RFC-2822 date
        -I[SPEC]        Output ISO-8601 date
                        SPEC=date (default), hours, minutes, seconds or ns

Recognized TIME formats:
        @seconds_since_1970
        hh:mm[:ss]
        [YYYY.]MM.DD-hh:mm[:ss]
        YYYY-MM-DD hh:mm[:ss]
        [[[[[YY]YY]MM]DD]hh]mm[.ss]
        'date TIME' form accepts MMDDhhmm[[YY]YY][.ss] instead
date: unrecognized option: rfc-3339=seconds
**Environment**:

### PR DESCRIPTION
**Pull request by jmickey**
**What this PR does / why we need it**:

#226 - Not all `date` commands are made equal. To resolve the issue we can specify the exact format of the date we want rather than use the `--rfc-3339` flag. There is a minor, but inconsequential difference in the output.

See below output:

GNU Date:

```
❯ gdate --rfc-3339=seconds | sed 's/ /T/'
2023-02-16T11:50:12+00:00
```

With explicit formatting:

```
❯ date -u +%FT%TZ
2023-02-16T11:51:20Z
```

Alternatively, we can remove the `Z` with:

```
❯ date -u +%FT%T%z
2023-02-16T11:52:38+0000
```

If the above is preferred I am happy to change it.

**Which issue(s) this PR fixes**:
Fixes #226

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,7 +14,7 @@ COMPONENT_CLI_IMAGE_REPOSITORY                 := $(REGISTRY)/cli
 SOURCES := $(shell go list -f '{{$$I:=.Dir}}{{range .GoFiles }}{{$$I}}/{{.}} {{end}}' ./... )
 GOPATH                                         := $(shell go env GOPATH)
 
-NOW         := $(shell date --rfc-3339=seconds | sed 's/ /T/')
+NOW         := $(shell date -u +%FT%TZ)
```
Comment (mandelsoft) [inline Makefile:17]: To generate the same result it should be `date  +%FT%T%:z`

Comment (jmickey) [reply Makefile]: Updated.

**Comment by jmickey** [comment]
@mandelsoft I've updated all references to `--rfc-3339` with that format.

**Comment by Skarlso** [comment]
Nice! This is a neat little fix. I can confirm that it's working on OSX:

```
 make
mkdir -p bin
go build -ldflags "-s -w -X github.com/open-component-model/ocm/pkg/version.gitVersion=v0.1.0+c26310b51e12a49dc68035b168c9874ef790c197 -X github.com/open-component-model/ocm/pkg/version.gitTreeState=clean -X github.com/open-component-model/ocm/pkg/version.gitCommit=c26310b51e12a49dc68035b168c9874ef790c197 -X github.com/open-component-model/ocm/pkg/version.buildDate=2023-02-16T13:26:49:z" -o bin/ocm ./cmds/ocm
go build -ldflags "-s -w -X github.com/open-component-model/ocm/pkg/version.gitVersion=v0.1.0+c26310b51e12a49dc68035b168c9874ef790c197 -X github.com/open-component-model/ocm/pkg/version.gitTreeState=clean -X github.com/open-component-model/ocm/pkg/version.gitCommit=c26310b51e12a49dc68035b168c9874ef790c197 -X github.com/open-component-model/ocm/pkg/version.buildDate=2023-02-16T13:26:49:z" -o bin/helminstaller ./cmds/helminstaller
go build -ldflags "-s -w -X github.com/open-component-model/ocm/pkg/version.gitVersion=v0.1.0+c26310b51e12a49dc68035b168c9874ef790c197 -X github.com/open-component-model/ocm/pkg/version.gitTreeState=clean -X github.com/open-component-model/ocm/pkg/version.gitCommit=c26310b51e12a49dc68035b168c9874ef790c197 -X github.com/open-component-model/ocm/pkg/version.buildDate=2023-02-16T13:26:49:z" -o bin/demo ./cmds/demoplugin
```

**Comment by Skarlso** [comment]
Ah this is not the same though as the gnu date:
```
version.buildDate=2023-02-16T13:26:49:z
``

gnu date:
```
gdate --rfc-3339=seconds | sed 's/ /T/'
2023-02-16T14:29:13+01:00
```

**Comment by jmickey** [comment]
MacOS `date` doesn't support `:z`. So it will need to just be `date -u +%FT%T%z`

**Comment by Skarlso** [comment]
```
date +%Y-%m-%dT%T%z
2023-02-16T15:46:45+0100
```

I think that's close enough. :)

**Comment by Skarlso** [comment]
I forgot to test this with linux:

```
gdate +%Y-%m-%dT%T%z
2023-02-16T18:11:50+0100
➜  replication-controller git:(conditions) date +%Y-%m-%dT%T%z
2023-02-16T18:11:54+0100
```

So GNU date will produce the same output. Nice! :)

**Review by Skarlso — APPROVED** [review]
LGTM

**Comment by mandelsoft** [comment]
I'm not sure why we need this at all. On MacOS nothing works correctly if you don't install the GNU utils. Therefore the installation of the GNU utils is a prerequisite,

see https://apple.stackexchange.com/questions/69223/how-to-replace-mac-os-x-utilities-with-gnu-core-utilities

Then you have to manipulate the PATH to use the new commands:
```
# GNU
export PATH="$(brew --prefix)/opt/coreutils/libexec/gnubin:$PATH"
export PATH="$(brew --prefix)/opt/findutils/libexec/gnubin:$PATH"
export PATH="$(brew --prefix)/opt/gnu-tar/libexec/gnubin:$PATH"
export PATH="$(brew --prefix)/opt/gnu-sed/libexec/gnubin:$PATH"
export PATH="$(brew --prefix)/opt/gnu-indent/libexec/gnubin:$PATH"
#export PATH="$(brew --prefix)/opt/gzip/bin:$PATH"
# MANPATH
export MANPATH="$(brew --prefix)/opt/coreutils/libexec/gnuman:MANPATH"
export MANPATH="$(brew --prefix)/opt/findutils/libexec/gnuman:MANPATH"
export MANPATH="$(brew --prefix)/opt/gnu-tar/libexec/gnuman:MANPATH"
export MANPATH="$(brew --prefix)/opt/gnu-sed/libexec/gnuman:MANPATH"
export MANPATH="$(brew --prefix)/opt/gnu-indent/libexec/gnuman:$MANPATH"
```

**Comment by Skarlso** [comment]
So the problematic part is that the script only installs tools, but you still have to do some manual spelunking to rename the local tools to the GNU ones.

OSX BSD tools however, are actually fine. They work differently, that is true, however, in this case, both commands:

osx and gnu
```
➜  date +%FT%T%z
2023-02-21T12:59:01+0100
➜  gdate +%FT%T%z
2023-02-21T12:59:05+0100
```

... will provide _the same value_. This means none of the above shenanigans will have to be used at all and the build process will work the same way on both environments without having to bother updating any of them. I think that's a win for convenience and consumer experience, right? :)

**Comment by jmickey** [comment]
This was discussed in the Wednesday sync. As it is a small change and doesn't introduce any breaking changes, but rather just makes it easier to run on systems without GNU tools, we decided it was best to simply merge the change.

**Pull request merged**